src/include/storage/arrow_block_metadata.h

```diff
@@ -0,0 +1,241 @@
+#pragma once
+#include <map>
+#include <unordered_set>
+#include "storage/block_layout.h"
+#include "storage/storage_defs.h"
+#include "storage/storage_util.h"
+
+namespace terrier::storage {
+
+// TODO(Tianyu): Can use to specify cases where we don't concat per-block in the future (e.g. no need to put
+//  blob into Arrow)
+/**
+ * Type of Arrow column
+ */
+enum class ArrowColumnType : uint8_t { FIXED_LENGTH = 0, GATHERED_VARLEN, DICTIONARY_COMPRESSED };
+
+/**
+ * Stores information about an Arrow varlen column. This class implements an Arrow list, with
+ * a byte array of values and an array of offsets into the value array
+ */
+class ArrowVarlenColumn {
+ public:
+  /**
+   * Constructs an empty Arrow VarlenColumn
+   */
+  ArrowVarlenColumn() = default;
+
+  /**
+   * Constructs a new arrow varlen column object
+   * @param values_length size of the values buffer, in bytes
+   * @param offsets_length number of elements in the offsets array
+   */
+  ArrowVarlenColumn(uint32_t values_length, uint32_t offsets_length)
```

Comment:
It's confusing to have one "length" arg mean bytes, and another mean "elements". Suggest renaming one or both.

Reply:
These are meant to be the length of these two Arrow arrays. Using Arrow's terminology for consistency. I modified 
the documentation of values to just be the length of the array, instead of the size in bytes, to hopefully make it clearer.